internal/core/server/workspace.go

```diff
@@ -704,8 +705,71 @@ func (ws *Workspace) DescribeProcesses(ctx context.Context, input *api.DescribeP
 				ID:       component.ID,
 				Name:     component.Name,
 				Provider: "docker",
-				Running:  state.Running,
 			}
+
+			containerInfo, err := ws.Docker.ContainerInspect(ctx, state.ContainerID)
+			if err == nil {
+				process.Running = containerInfo.State.Running
+				if process.Running {
+					statRequest, err := ws.Docker.ContainerStats(ctx, state.ContainerID, false)
+					if err != nil {
+						return nil, fmt.Errorf("could not get container stats: %w", err)
+					}
+					defer statRequest.Body.Close()
+
+					scanner := bufio.NewScanner(statRequest.Body)
+					didScan := scanner.Scan()
+					if err = scanner.Err(); err != nil {
+						return nil, fmt.Errorf("error scanning container stats: %w", err)
+					}
+
+					if !didScan {
+						return nil, fmt.Errorf("could not read container stats")
+					}
+
+					var containerStats DockerStats
+					err = json.Unmarshal(scanner.Bytes(), &containerStats)
+					if err != nil {
+						return nil, fmt.Errorf("could not unmarshal container stats: %w", err)
+					}
+
+					process.ResidentMemory = containerStats.MemoryStats.Usage
+					layout := "2006-01-02T15:04:05.000000000Z"
+					startTime, err := time.Parse(layout, containerInfo.State.StartedAt)
+					if err != nil {
+						return nil, fmt.Errorf("could not parse start time: %w", err)
+					}
+
+					process.CreateTime = startTime.UnixNano() / 1e6
+
+					if containerStats.CPUStats.SystemCPUUsage != 0 {
+						process.CPUPercent = float64(containerStats.CPUStats.CPUUsage.TotalUsage) / 1e9
+					}
+
+					process.EnvVars = map[string]string{}
+					for _, env := range containerInfo.Config.Env {
+						decomposedEnv := strings.SplitN(env, "=", 2)
+						process.EnvVars[decomposedEnv[0]] = decomposedEnv[1]
+					}
+
+					process.Ports = []uint32{}
```

Comment:
When you have a known array length that you're copying/mapping, use this pattern:

```go
process.Ports = make([]uint32, len(containerInfo.NetworkSettings.Ports)
for i, p := range containerInfo.NetworkSettings.Ports {
  processPorts[i] = ...
}
```